client/js/widgets/widget.js

```diff
@@ -1387,6 +1431,11 @@ export class Widget extends StateManaged {
             }
           } else {
             for(const w of collections[collection]) {
+              if (readOnlyProperties.has(a.property) || w.isTypeSpecificReadOnlyProperty(a.property)) {
```

Comment:
How about instead adding a `readOnlyProperties()` function similar to `cssProperties()` which subclasses can use to specify more read only properties? Even better would be something like an internalProperties map to the function which calculates that property value but that's a bit outside of the scope of this change.

Reply:
Not a bad idea, but all of this is out of scope for this PR. `readOnlyProperties` was added some time back; all I did was add the type specific portion.

Reply:
I agree that the internalProperties map is out of scope, but this is the first patch to add type specific read only properties so it makes sense to align it with the way type specific css properties work.

All that's needed is to add the `readOnlyPropertes()` function to widget and override it in scoreboard to add the extra read only property and then use this function instead of directly using the local constant.

Reply:
Something like this:
In widget.js
```js
const widgetReadOnlyProperties = new Set([
  '_absoluteRotation',
  '_absoluteScale',
  '_absoluteX',
  '_absoluteY',
  '_ancestor',
  '_centerAbsoluteX',
  '_centerAbsoluteY',
  '_localOriginAbsoluteX',
  '_localOriginAbsoluteY'
]);

export class Widget extends StateManaged {
...
readOnlyProperties() {
  return widgetReadOnlyProperties;
}
```

And then in scoreboard.js you could do something like:
```js
readOnlyProperties() {  
  return new Set([...super.readOnlyProperties(), '_totals']);
}
```

Admittedly it would be cleaner to set up a single set statically for the scoreboard widget which could be done by calling the widget method to set up a static local like this:

```js
const scoreboardReadOnlyProperties = new Set(
    [...Widget.prototype.readOnlyProperties.apply(null),  '_totals']);

readOnlyProperties() {  
  return scoreboardReadOnlyProperties;
}
```

This way we don't have to remember to check the type specific read only properties set anywhere that the read only properties set is checked.

In widget.js you can throw an error in get() if the property is in readOnlyProperties and doesn't match one in the list because the superclass that added it should also have overridden the getter for that property, i.e. this line:
https://github.com/ArnoldSmith86/virtualtabletop/blob/main/client/js/widgets/widget.js#L1627

Reply:
Thanks.

Any ideas on my css problem?

Reply:
Done.